icu4c/source/common/umutex.cpp

```diff
@@ -35,34 +36,98 @@ U_NAMESPACE_BEGIN
 #error U_USER_MUTEX_CPP not supported
 #endif
 
+// Check that UMutex is trivially constructable & destructable, which ensures that
+// static instances are not running static constructors or destructors.
+#if 0    // Omit because some build bot compilers are old, and lack support for the test.
```

Comment:
I wonder if maybe this could be skipped for older compilers?
`
#if (defined(__GNUG__) && __GNUC__ < 5) || (defined(__clang__) && __clang_major__ < 5)
// skip
#else
// test
#endif
`